### ISSUE TITLE
bazel_4 fails to build on aarch64-darwin

### ISSUE THREAD

**Issue by uri-canva** [NixOS/nixpkgs#134758]
### Describe the bug
Example build failure: https://hydra.nixos.org/build/150263034/nixlog/1

### Steps To Reproduce
1. Build `nixpkgs.bazel_4`

### Expected behavior
Builds correctly.

### Notify maintainers
@mboes

### Metadata
 - system: `"aarch64-darwin"`
 - host os: `Darwin 20.6.0, macOS 11.5.2`
 - multi-user?: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixpkgs-21.11.pre309670.253aecf69ed"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by uri-canva** [comment]
Couldn't reproduce failure on:
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixpkgs-21.11pre299952.7918dc5148d"`
 - channels(uri): `"darwin"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by veprbl** [comment]
Upstream claims support M1 in v4.1 https://github.com/bazelbuild/bazel/issues/13099#issuecomment-823333720
Looks like an incompatible argument is supplied to ar

**Comment by uri-canva** [comment]
According to https://github.com/bazelbuild/bazel/releases/tag/4.1.0 the cherry picks requested in that comment were accepted in that version.

**Comment by uri-canva** [comment]
`4.2.0` has a different m1 fix: https://github.com/bazelbuild/bazel/issues/13514, might be worth a shot.

**Comment by domenkozar** [comment]
https://github.com/NixOS/nixpkgs/pull/136004 if someone wants to try

**Comment by divanorama** [comment]
There seems to be one more issue in hydra https://hydra.nixos.org/build/152904307/nixlog/1
```console
checking for references to /build/ in /nix/store/zpwcnd90shsvjrfc8nj88mz79vn0mqim-remote_java_tools_linux...
automatically fixing dependencies for ELF files
searching for dependencies of /nix/store/zpwcnd90shsvjrfc8nj88mz79vn0mqim-remote_java_tools_linux/java_tools/ijar/zipper
  libstdc++.so.6 -> not found!
searching for dependencies of /nix/store/zpwcnd90shsvjrfc8nj88mz79vn0mqim-remote_java_tools_linux/java_tools/ijar/ijar
  libstdc++.so.6 -> not found!
searching for dependencies of /nix/store/zpwcnd90shsvjrfc8nj88mz79vn0mqim-remote_java_tools_linux/java_tools/src/tools/singlejar/singlejar_local
  libstdc++.so.6 -> not found!
  ld-linux-x86-64.so.2 -> not found!
autoPatchelfHook could not satisfy dependency ld-linux-x86-64.so.2 wanted by /nix/store/zpwcnd90shsvjrfc8nj88mz79vn0mqim-remote_java_tools_linux/java_tools/src/tools/singlejar/singlejar_local
autoPatchelfHook could not satisfy dependency libstdc++.so.6 wanted by /nix/store/zpwcnd90shsvjrfc8nj88mz79vn0mqim-remote_java_tools_linux/java_tools/src/tools/singlejar/singlejar_local
Add the missing dependencies to the build inputs or set autoPatchelfIgnoreMissingDeps=true
builder for '/nix/store/vfjl0fc75vqv0nln6jr1zmphi3dkwmvl-remote_java_tools_linux.drv' failed with exit code 1
```
propagated from 4.1.0 to 4.2.1

**Comment by cocreature** [comment]
I seem to be getting a different error
```
FAIL: //examples/cpp:hello-success_test (see /private/tmp/nix-build-bazel-4.2.1.drv-0/_bazel_builder/be75a0ea28bc829c1f56b23f6947b302/execroot/io_bazel/bazel-out/darwin_arm64-fastbuild/testlogs/examples/cpp/hello-success_test/test.log)
[19 / 25] 2 actions running
    @bazel_tools//tools/jdk:platformclasspath; 0s processwrapper-sandbox
    Testing //examples/cpp:hello-success_test; 0s processwrapper-sandbox
INFO: From Testing //examples/cpp:hello-success_test:
[19 / 25] 1 / 2 tests, 1 failed; 2 actions running; last test: ...success_test
    @bazel_tools//tools/jdk:platformclasspath; 0s processwrapper-sandbox
    Testing //examples/cpp:hello-success_test; 0s processwrapper-sandbox
==================== Test output for //examples/cpp:hello-success_test:
================================================================================
[19 / 25] 1 / 2 tests, 1 failed; 2 actions running; last test: ...success_test
    @bazel_tools//tools/jdk:platformclasspath; 0s processwrapper-sandbox
    Testing //examples/cpp:hello-success_test; 0s processwrapper-sandbox
[21 / 25] 1 / 2 tests, 1 failed;  1 action; last test: ...p:hello-success_test
    @bazel_tools//tools/jdk:platformclasspath; 0s processwrapper-sandbox
[22 / 25] 1 / 2 tests, 1 failed; 2 actions, 0 running; last test: ...cess_test
    [Prepa] Building .../example/myproject/libhello-lib.jar (1 source file)
    [Prepa] .../java-native/src/main/java/com/example/myproject:hello-lib
[23 / 25] 1 / 2 tests, 1 failed; 2 actions, 1 running; last test: ...cess_test
    .../src/main/java/com/example/myproject:hello-lib; 0s multiplex-worker
    [Prepa] Building .../java/com/example/myproject/hello.jar (1 source file)
[23 / 25] 1 / 2 tests, 1 failed; 2 actions running; last test: ...success_test
    .../src/main/java/com/example/myproject:hello-lib; 0s multiplex-worker
    Building .../myproject/hello.jar (1 source file); 0s multiplex-worker
[25 / 26] 1 / 2 tests, 1 failed;  1 action; last test: ...p:hello-success_test
    [Prepa] Testing //.../src/test/java/com/example/myproject:hello
[25 / 26] 1 / 2 tests, 1 failed;  1 action; last test: ...p:hello-success_test
    Testing //.../java/com/example/myproject:hello; 0s processwrapper-sandbox
INFO: Elapsed time: 9.754s, Critical Path: 2.62s
[26 / 26] 2 / 2 tests, 1 failed; checking cached actions; last test: ...:hello
INFO: 26 processes: 11 internal, 13 processwrapper-sandbox, 2 worker.
[26 / 26] 2 / 2 tests, 1 failed; checking cached actions; last test: ...:hello
INFO: Build completed, 1 test FAILED, 26 total actions
//examples/java-native/src/test/java/com/example/myproject:hello         PASSED in 0.3s
//examples/cpp:hello-success_test                                        FAILED in 0.1s
  /private/tmp/nix-build-bazel-4.2.1.drv-0/_bazel_builder/be75a0ea28bc829c1f56b23f6947b302/execroot/io_bazel/bazel-out/darwin_arm64-fastbuild/testlogs/examples/cpp/hello-success_test/test.log
```
so it looks like everything built fine but then it fails to run the install tests. The log just seems empty so no idea what is failing.

`doInstallCheck` gets me a bit further. Not yet at the point where I can say how usable it is.

**Comment by liji-canva** [comment]
I had the same `//examples/cpp:hello-success_test` error. I found out that the binary built from the `cc_test` returns with error code 137.  But I don't have enough knowledge to further debug it.

**Comment by winterqt** [comment]
I'm going to try to tackle this.

**Comment by winterqt** [comment]
cc @NixOS/darwin-maintainers, please see the above comment if you're on `aarch64-darwin` and are running Big Sur.

**Comment by uri-canva** [comment]
I have Monterey and I can reproduce the `protoc` error in https://github.com/NixOS/nixpkgs/issues/134758#issuecomment-940384499.
@liji-canva has Big Sur and cannot reproduce it, it builds correctly for him (minus the tests).

**Comment by uri-canva** [comment]
Hydra doesn't run Monterey yet (it doesn't even run Big Sur), so let's disable `doInstallCheck` on `aarch64-darwin`, and people running Monterey will be able to download `bazel_4` from the nix binary cache.
I've noticed this issue with some other packages as well: they cannot be built on newer versions of macOS, but downloading them from hydra and running them is all good, I think a lot of those issues are the same @toonn is hitting in the SDK update.

**Comment by uri-canva** [comment]
Split out the install check failure into a separate issue: https://github.com/NixOS/nixpkgs/issues/145587

**Comment by uri-canva** [comment]
Split out the Monterey build failure out in a separate issue: https://github.com/NixOS/nixpkgs/issues/145590

**Comment by uri-canva** [comment]
Sorry for the issue fragmentation, but I believe it will help keep the discussions more focused and make it easier for people experiencing the issues to find them.

I will consider this issue solved once https://github.com/NixOS/nixpkgs/pull/145588 is merged, and we can continue debugging the other issues separately.

**Comment by uri-canva** [comment]
@winterqt can you open a PR for the md5sum issue in https://github.com/NixOS/nixpkgs/issues/134758#issuecomment-965885763?

**Comment by winterqt** [comment]
> @winterqt can you open a PR for the md5sum issue in [#134758 (comment)](https://github.com/NixOS/nixpkgs/issues/134758#issuecomment-965885763)?

@uri-canva Sure, opened: #145709.
